### PR TITLE
Explicitly cast invalid values to suppress warnings

### PR DESCRIPTION
**Pull request by kalj**
@kronbichler: This gives warnings when compiling with NVCC.

### CODE REVIEW

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -1800,10 +1800,10 @@ ::FEEvaluationBase (const Mapping<dim>       &mapping,
                     const unsigned int        first_selected_component,
                     const FEEvaluationBase<dim,n_components_other,Number> *other)
   :
-  quad_no            (-1),
+  quad_no            (static_cast<unsigned int>(-1)),
   n_fe_components    (n_components_),
-  active_fe_index    (-1),
-  active_quad_index  (-1),
+  active_fe_index    (static_cast<unsigned int>(-1)),
+  active_quad_index  (static_cast<unsigned int>(-1)),
```
Comment (bangerth) [inline include/deal.II/matrix_free/fe_evaluation.h]: @kronbichler -- shouldn't all of these be `numbers::invalid_unsigned_int`? Or do you somewhere test them against `-1`?

**Comment by kalj** [comment]
I agree with @bangerth. I'll repush once ok'd.

**Comment by kronbichler** [comment]
I agree with this patch @kalj. The value should be `numbers::invalid_unsigned_int` as mentioned by @bangerth (this way, any unguarded access in release mode will produce a segmentation fault, but we do not check for -1 explicitly).

**Comment by kalj** [comment]
Patch updated.

**Pull request merged**